File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-8.cpp
#include<ranges>
class Solution {
public:
    int characterReplacement(string s, int k) {
           unordered_map<char, int> count;
           int longest{0}, l{0}, maxfqt{0};
           for(auto&& [r, c]: std::views::enumerate(s)){
                count[c]++;
                maxfqt= max(maxfqt, count[c]);
                while (static_cast<int>(r)-l+1-maxfqt>k){
                    count[s[l]]--;
                    l++;
                }

                longest=max(static_cast<int>(r)-l+1, longest);
           }
           
            return longest;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-8.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        rs=''
        maxcount=0
        for c in s:
            if c not in rs:
                rs= rs+c
                maxcount=max(maxcount, len(rs))
            else:
                rs=rs[rs.index(c)+1:]+c

        return maxcount